src/toil_lib/programs.py

```diff
@@ -55,6 +68,12 @@ def docker_call(tool,
     if outputs is None:
         outputs = {}
 
+    # Docker does not allow the --rm flag to be used when the container is run in detached mode.
+    require(not (rm and detached), "Conflicting options 'rm' and 'detached'.")
+    # Ensure the user has passed a valid value for defer
+    require(defer in (None, docker_call.FORGO, docker_call.STOP, docker_call.RM), 'Please provide '
+            'a valid value for defer.')
```

Comment:
I'd prefer not wrapping strings and just keeping them whole on a line by themselves unless that increases the number of lines overall.